filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -169,7 +169,6 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 		cim.Logger.Warn("filestream input without ID is discouraged, please add an ID and restart Filebeat")
 	}
 
-	metricsID := settings.ID
```

Comment:
It'll cause the metrics for an input without ID to be published under the generated ID from the context. As it's the same behaviour as some other inputs have, I chose to allow it and avoid having filestream managing the metrics registry "manually". That way filestream does not need to explicitly remove the metrics registry from the http monitoring endpoint and unregister from the namespace where it's created. Right now the namespace is the global `dataset` namaspace.